### PR TITLE
DB-test

### PR DESCRIPTION
**Pull request by akoniar**

### CODE REVIEW

**Comment by spisiakm** [comment]
@cescoffier if you have some spare time, please just have a look. Thanks !

File: various-db-engines-it/db-it/mysql-it/src/test/java/io/vertx/openshift/mysql/MySQLInternalDbIT.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import static io.vertx.it.openshift.utils.Kube.awaitUntilPodIsReady;
+
+public class MySQLInternalDbIT extends AbstractInternalDBTestClass {
+
+  @BeforeClass
+  public static void initialize() throws IOException {
+    System.out.println("Deploying mysql");
+
+    OC.execute("project", client.getNamespace());
+    OC.execute("new-app", "registry.access.redhat.com/rhscl/mysql-57-rhel7",
```
Comment (cescoffier) [inline various-db-engines-it/db-it/mysql-it/src/test/java/io/vertx/openshift/mysql/MySQLInternalDbIT.java]: is this the supported one (official) ?

File: various-db-engines-it/db-it/mysql-it/src/test/java/io/vertx/openshift/mysql/MySQLInternalDbIT.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+    awaitUntilPodIsReady(client, DB_NAME);
+    deployAndAwaitStartWithRoute("/healthcheck");
+  }
+
+  @Override
+  protected void deployDB (){
+    OC.execute("project", client.getNamespace());
+    OC.execute("new-app", "registry.access.redhat.com/rhscl/mysql-57-rhel7",
```
Comment (cescoffier) [inline various-db-engines-it/db-it/mysql-it/src/test/java/io/vertx/openshift/mysql/MySQLInternalDbIT.java]: why do we have it twice?

File: various-db-engines-it/db-it/oracle-it/pom.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    <artifactId>db-it</artifactId>
+    <version>1.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>oracle-it</artifactId>
+
+  <properties>
+    <vertx.verticle>io.vertx.openshift.oracledb.OracleVerticle</vertx.verticle>
+    <vertx.health.path>/healthcheck</vertx.health.path>
+    <nexus.repo>http://www.maven.example.com/</nexus.repo>
```
Comment (cescoffier) [inline various-db-engines-it/db-it/oracle-it/pom.xml]: Is it required?

File: various-db-engines-it/db-it/oracle-it/src/main/java/io/vertx/openshift/oracledb/OracleVerticle.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    store.create(item)
+      .subscribe(
+        json -> ((JdbcOracleVegetableStore) store).read(json.getString("rowId"))
+          .subscribe(entries -> ctx.response()
+            .putHeader("Location", "/api/vegetables/" + json.getLong("id"))
+            .putHeader("Content-Type", "application/json")
+            .setStatusCode(201)
+            .end(entries.encodePrettily())),
+        err -> {
+          System.out.println(err.toString());
```
Comment (cescoffier) [inline various-db-engines-it/db-it/oracle-it/src/main/java/io/vertx/openshift/oracledb/OracleVerticle.java]: Remove trace if not required

File: various-db-engines-it/db-it/oracle-it/src/main/resources/db_init.sql [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  id        NUMBER GENERATED ALWAYS as IDENTITY(START with 1 INCREMENT by 1),
+  name      VARCHAR2(40) NOT NULL,
+  amount    INT,
+  CONSTRAINT vegetables_pk PRIMARY KEY (id));
+
+DELETE FROM vegetables;
+
+INSERT INTO vegetables (name, amount) values ('Carrots', 10);
+INSERT INTO vegetables (name, amount) values ('Tomatoes', 10);
+INSERT INTO vegetables (name, amount) values ('Onions', 10);
```
Comment (cescoffier) [inline various-db-engines-it/db-it/oracle-it/src/main/resources/db_init.sql]: Are onions a vegeatable? Ok it is.... https://en.wikipedia.org/wiki/Onion

File: various-db-engines-it/db-it/oracle-it/src/test/java/io/vertx/openshift/oracledb/OracleExternalDbIT.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package io.vertx.openshift.oracledb;
+
+import io.vertx.openshift.utils.AbstractExternalDBTestClass;
+
+/**
+ * @author Martin Spisiak (mspisiak@redhat.com) on 19/10/17.
+ */
+public class OracleExternalDbIT extends AbstractExternalDBTestClass {
+
+
```
Comment (cescoffier) [inline various-db-engines-it/db-it/oracle-it/src/test/java/io/vertx/openshift/oracledb/OracleExternalDbIT.java]: Shouldn't it do something?

File: various-db-engines-it/db-it/postgresql-it/src/test/java/io/vertx/openshift/postgresql/PostgreSQLExternalDbIT.java [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package io.vertx.openshift.postgresql;
+
+import io.vertx.openshift.utils.AbstractExternalDBTestClass;
+
+public class PostgreSQLExternalDbIT extends AbstractExternalDBTestClass {
+
```
Comment (cescoffier) [inline various-db-engines-it/db-it/postgresql-it/src/test/java/io/vertx/openshift/postgresql/PostgreSQLExternalDbIT.java]: Add comment to explain why it's empty.

File: various-db-engines-it/db-it/oracle-it/src/test/java/io/vertx/openshift/oracledb/OracleExternalDbIT.java (hunk 6 above)
Comment (cescoffier) [reply various-db-engines-it/db-it/oracle-it/src/test/java/io/vertx/openshift/oracledb/OracleExternalDbIT.java]: It looks like it's expected to be empty. Please add a comment explaining why.

File: various-db-engines-it/pom.xml [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+    <dependency>
+      <groupId>io.vertx.openshift</groupId>
+      <artifactId>vertx-it-utils</artifactId>
+      <version>${project.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>mysql</groupId>
+      <artifactId>mysql-connector-java</artifactId>
+      <version>5.1.39</version>
```
Comment (cescoffier) [inline various-db-engines-it/pom.xml]: Are these versions the officially supported ones? (for all drivers)

File: various-db-engines-it/verticle-utils/src/main/java/io/vertx/openshift/utils/AbstractInternalDBTestClass.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+public abstract class AbstractInternalDBTestClass extends AbstractDBTestClass {
+
+  @Test
+  public void runDBAfterDeployAppTest () throws Exception {
+    System.out.println("Run deploy app before deploy db");
+    dbCleanup();
+    cleanup();
+    deploymentAssistant.deployApplication();
+    TimeUnit.SECONDS.sleep(10);
```
Comment (cescoffier) [inline various-db-engines-it/verticle-utils/src/main/java/io/vertx/openshift/utils/AbstractInternalDBTestClass.java]: can't we wait for readiness?

File: various-db-engines-it/db-it/oracle-it/pom.xml (hunk 3 above)
Comment (spisiakm) [reply various-db-engines-it/db-it/oracle-it/pom.xml]: well I thought that if we want to automate these tests (i.e. using our xPaaS Jenkins instance), then we need to have a way to get the Oracle jdbc driver. One would be through this property, as we have an internal nexus repo containing Oracle jdbc driver, so I would just configure this property when running automated tests.

**Comment by cescoffier** [comment]
Thanks, great job!

**Pull request merged**